File: example/features/step_definitions/calculator_step.rb
# encoding: utf-8
require 'spec/expectations'
$:.unshift(File.dirname(__FILE__) + '/../../lib') # This line is not needed in your own project
require 'cucumber/formatters/unicode'
require 'calculator'

Before do
  @calc = Calculator.new
end

After do
end

Given /I have entered (\d+) into the calculator/ do |n|
  @calc.push n.to_i
end

When /I press (\w+)/ do |op|
  @result = @calc.send op
end

Then /the result should be (.*) on the screen/ do |result|
  @result.should == result.to_f * 10
end

Then /the result should be a number/ do
  @result.should be_instance_of(Fixnum)
end

File: example/features/step_definitions/calculator_step.py
import sys, os
sys.path.append(os.getcwd())
import pyeature

import re

import calculator

def before():
    self.calc = calculator.Calculator()


@pyeature.given(re.compile("I have entered ([0-9]+) into the calculator"))
def enter_number():
    num = args[1]
    self.calc.append(int(num))

def when_i_press_add():
    self.result = self.calc.add()

@pyeature.then("the result should be 120 on the screen")
#def then_the_result_should_be_120_on_the_screen():
def check_result():
    assert self.result == 1200

def then_xxthe_result_should_be_a_number():
    assert type(self.result) is int

